FlashSpace/Features/Workspaces/WorkspaceManager.swift

```diff
@@ -87,11 +87,22 @@ final class WorkspaceManager: ObservableObject {
             .sink { [weak self] application in
                 guard let self else { return }
 
+                // Track focused app for the workspace it belongs to
                 if let activeWorkspace = activeWorkspace[application.display ?? ""],
                    activeWorkspace.apps.containsApp(application) {
                     lastFocusedApp[activeWorkspace.id] = application.toMacApp
                     updateActiveWorkspace(activeWorkspace)
                 }
+
+                // Also track floating apps for workspaces on the same display
+                if floatingAppsSettings.floatingApps.containsApp(application) {
+                    let displayName = application.display ?? ""
+
+                    // Store the floating app as the last focused app for all workspaces on the same display
+                    for workspace in activeWorkspace.values where workspace.displayWithFallback == displayName {
+                        lastFocusedApp[workspace.id] = application.toMacApp
+                    }
```

Comment:
```suggestion
                    activeWorkspace.values
                        .filter { $0.displayWithFallback == display }
                        .forEach { lastFocusedApp[$0.id] = application.toMacApp }
```

Reply:
I'm not sure about updating all workspaces. I think only the active workspace should be updated.